packages/block-editor/src/components/inserter/index.js

```diff
@@ -426,13 +424,7 @@ export const ComposedPrivateInserter = compose( [
 ] )( PrivateInserter );
 
 const Inserter = forwardRef( ( props, ref ) => {
-	return (
-		<ComposedPrivateInserter
-			ref={ ref }
-			{ ...props }
-			orderInitialBlockItems={ undefined }
```

Comment:
I suppose the other option is to keep this API, but instead support passing a custom appender to List View (instead of the showAppender prop).

It still doesn't solve the issue where the block itself ends up with different ordering, unless both the block and the list view instance use the same custom appender, so I think this is better. It's define once and it works everywhere which means there's less room for errors.

Reply:
I think 

> instead support passing a custom appender to List View (instead of the showAppender prop).

is better than the prop, but why would we need to keep `orderInitialBlockItems`?

Reply:
Only if you wanted _specific_ instance X of an inserter to have a specific order.

The new implementation in this PR is a blanket application. Any block that uses the new prop on `InnerBlocks` no matter where it is with have items prioritised.

The previous implementation was more targetted.

Reply:
Wouldn't `orderInitialBlockItems` be the default order for when block's don't prioritize or for the generic canvas inserter in an empty paragraph?